### ISSUE TITLE
TypeError: Cannot read property 'hasOwnProperty' of undefined (verb 0.8.3, verb-cli 0.7.0)

### ISSUE THREAD

**Issue by nknapp** [verbose/verb#62]
Hi,

in recent versions (0.8.x) of verb there seems to be a problem that causes an exception for all of my repositories. I've reduced the verbfile.js to

```js
var verb = require('verb')

verb.task('default', function () {
  verb.src(['.verb.md'])
    .pipe(verb.dest('./'))
})
```
and I have an empty `.verb.md` file and still get the following exception:

```
TypeError: Cannot read property 'on' of undefined
    at proxyStream (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/event-stream/node_modules/duplexer/index.js:65:15)
    at Array.forEach (native)
    at forEach (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/event-stream/node_modules/duplexer/index.js:11:20)
    at duplex (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/event-stream/node_modules/duplexer/index.js:27:5)
    at Object.module.exports (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/event-stream/node_modules/stream-combiner/index.js:8:17)
    at createStack (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/lib/stack.js:96:24)
    at Object.exports.src (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/lib/stack.js:35:10)
    at Verb.src (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/index.js:65:16)
    at Verb.<anonymous> (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/verbfile.js:22:8)
    at module.exports (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/orchestrator/lib/runTask.js:34:7)
```
No problems with verb@0.7.4

**Comment by tunnckoCore** [comment]
try to update `verb-cli` to lastest and report back?

**Comment by nknapp** [comment]
```
[21:35:38] CLI version 0.7.0
[21:35:38] Local version 0.8.3
/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/lib/transforms/init/load.js:29
  if (paths.hasOwnProperty(name)) {
           ^
TypeError: Cannot read property 'hasOwnProperty' of undefined
    at opts (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/lib/transforms/init/load.js:29:12)
    at Verb.module.exports (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/lib/transforms/init/load.js:16:25)
    at Verb.Template.transform (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/template/index.js:272:8)
    at Verb.<anonymous> (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/lib/init.js:54:9)
    at Verb.on (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/template/node_modules/config-cache/node_modules/component-emitter/index.js:63:8)
    at Verb.Emitter.emit (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb/node_modules/template/node_modules/config-cache/node_modules/component-emitter/index.js:131:20)
    at Liftoff.run (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb-cli/bin/verb.js:124:12)
    at Liftoff.<anonymous> (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb-cli/node_modules/liftoff/index.js:192:16)
    at module.exports (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb-cli/node_modules/liftoff/node_modules/flagged-respawn/index.js:17:3)
    at Liftoff.<anonymous> (/home/nknappmeier/privat/node-libraries/multilang-extract-comments/node_modules/verb-cli/node_modules/liftoff/index.js:185:9)
```
Those are the current `master` branches of both modules. Same exception if I use the current versions from npm.

**Comment by tunnckoCore** [comment]
mm yea same here, confirmed.

@jonschlinkert mark it as bug, i'll review it this night and try to help

**Comment by jonschlinkert** [comment]
but verb-cli is on 0.7. have you tried that version?

**Comment by jonschlinkert** [comment]
> i'll review it this night and try to help

great thanks!

**Comment by nknapp** [comment]
@tunnckoCore: Thanks for looking into it. I don't know why, but when I tried this first (two days ago) I really thought I had done `npm -g install verb-cli` before. Somehow, I was still using 0.6.2 although 0.7.0 was already published for some days.

I've updated the issue name to match the exception that comes with 0.7.0, just in case someone else is looking for it.

**Comment by jonschlinkert** [comment]
did you also try `npm -g cache clean`?

**Comment by nknapp** [comment]
not until @tunnckoCore [told me to try the latest `verb-cli` version](#issuecomment-104937201). I have some questions or suggestions here, but I'll open a new issue for that. It's a bit off-topic here.

**Comment by jonschlinkert** [comment]
> I have some questions or suggestions here,

great, would love to hear them! I'll be on the lookout for the issues

**Comment by nknapp** [comment]
The issue seems to be resolved with verb 0.8.7. Thanks.
I think 0.8.7 is not on npm,yet. Is that possible? Could you publish it?

**Comment by jonschlinkert** [comment]
> I think 0.8.7 is not on npm,yet. Is that possible? Could you publish it?

really? strange I have a workflow for publishing.. not sure how that happened. I'll get it pushed up
thx for letting me know

**Comment by nknapp** [comment]
Mhm, this could be an issue with npm: http://registry.npmjs.org/verb shows a package `"0.8.7": "2015-05-28T10:21:39.541Z"`. But http://registry.npmjs.org/verb/0.8.7 returns `{"error":"version not found: 0.8.7"}` while http://registry.npmjs.org/verb/0.8.6 clearly returns the registry-entry for version 0.8.6

But [https://www.npmjs.com/package/verb](https://www.npmjs.com/package/verb) shows 
```
jonschlinkert jonschlinkert published 3 days ago
0.8.6 is the latest of 77 releases
github.com/assemble/verb
MIT license
```
And if I do `npm install verb', even after a `npm -g cache clean` and after wiping the `node_modules` folder, I still get version 0.8.6.

May be some inconsistencies between npm-mirrors. Lags in couchdb-replication? But I doubt that it really takes that long for the mirrors to synchronize.

**Comment by nknapp** [comment]
I'm closing this issue, since it is resolved.  Not having the newest version on npm is a different thing that may not be an issue of verb

**Comment by jonschlinkert** [comment]
ah, wait - I remember.... I uninstalled that version because right after I pushed up I noticed a potential bug  and wanted to get it fixed first

**Comment by stefanwalther** [comment]
If you use the latest version `npm install verbose/verb --save-dev` it work. But the probably the bug @jonschlinkert is talking about is not fixed ;-)

**Comment by adjohnson916** [comment]
:+1:

@jonschlinkert, whenever you have time to push 0.8.7, it definitely fixes this issue occurring with 0.8.6 using with verb-cli@0.7.2 for me.

**Comment by tunnckoCore** [comment]
> occurring with 0.8.6 using with verb-cli@0.7.2 for me.

same here.

**Comment by jonschlinkert** [comment]
great to know, thanks! I'll get this updated asap. fwiw I have a bunch of other changes to push up, and I refactored [template](https://github.com/jonschlinkert/template) to speed things up in verb... so I'll try to get all of that pushed up by this weekend

**Comment by tunnckoCore** [comment]
> and I refactored template

good to hear, we'll stay tuned.

**Comment by adjohnson916** [comment]
@jonschlinkert Just a friendly reminder that pushing a 0.8.7 version would be great to fix this.

**Comment by adjohnson916** [comment]
FYI @jonschlinkert has released `verb@0.8.8` and `verb-cli@0.7.3`. These latest versions appear to be compatible with each other. I did notice an incompatibility between `verb@0.8.8` and `verb-cli@0.7.2`, so be sure to update both.
To minimize compatibility issues, I'd also recommend you install `verb-cli` locally to your project, rather than globally.
```sh
npm i --save-dev verb-cli@0.7.3 verb@0.8.8
```
Then you might want add an npm script to run verb, which will use your project's version of `verb-cli`. In your `package.json`:
```json
{
  "scripts": {
    "verb": "verb"
  }
}
```
Then run verb with:
```sh
npm ru verb
```

**Comment by jonschlinkert** [comment]
> I did notice an incompatibility between verb@0.8.8 and verb-cli@0.7.2, so be sure to update both.

hmm, you're right. perhaps we should un-release both patches and re-release as minor bumps. normally that would be odd but given that both are probably used standalone and not being consumed as a lib in another application, it will probably be ok. thoughts?

**Comment by jonschlinkert** [comment]
@adjohnson916 btw thank you for being on top of this!

**Comment by adjohnson916** [comment]
@jonschlinkert Minor bumps to replace the patches would probably be better. This migbt happen again though. Ideally we'd have some sort of test suite that could build a compatibility matrix, and the two projects could check for this or at least refer to it on any unexpected errors, then more helpfully inform the user.

**Comment by jonschlinkert** [comment]
> Ideally we'd have some sort of test suite that could build a compatibility matrix

agreed, tests are a big part of the next release. 

> pull more logic out of the CLI into the core, or perhaps formalize an more extensible interface between them.

I was just going to respond something to this effect lol. I'll try to push up a baseline for the next release so we can discuss possibilities for this.
